_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,194 @@
+
+```bash
+$ kubectl logs $(kubectl get pods -o name -n openfaas | grep -m1 kafka-connector | cut -d'/' -f 2) -n openfaas --tail 100 -f
+```
+
+In the first terminal write the message you would like your function to receive and check the logs output.
+
+```
+...
+2019/03/16 10:08:26 Syncing topic map
+[#1] Received on [faas-request,0]: 'Test the function.'
+2019/03/16 10:08:26 Invoke function: figlet
+2019/03/16 10:08:27 connector-sdk got result: [200] faas-request => figlet (480) bytes
+[200] faas-request => figlet
+ _____         _     _   _             __                  _   _               
+|_   _|__  ___| |_  | |_| |__   ___   / _|_   _ _ __   ___| |_(_) ___  _ __    
+  | |/ _ \/ __| __| | __| '_ \ / _ \ | |_| | | | '_ \ / __| __| |/ _ \| '_ \   
+  | |  __/\__ \ |_  | |_| | | |  __/ |  _| |_| | | | | (__| |_| | (_) | | | |_ 
+  |_|\___||___/\__|  \__|_| |_|\___| |_|  \__,_|_| |_|\___|\__|_|\___/|_| |_(_)
+                                                                               
+
+2019/03/16 10:08:29 Syncing topic map
+2019/03/16 10:08:32 Syncing topic map
+2019/03/16 10:08:35 Syncing topic map
+...
+```
+
+## But why do I need this combination?
+
+There are multiple real world examples where Kafka in combination with the Serverless approach provided by OpenFaaS becomes really powerful tool as Kafka can make even the smallest of systems extensible and OpenFaaS provides us with the quick way to add those extensions.
+
+Lets say you need to process customer information. When customer applies for you application you need to process it and store the information in a database and there are tons of customers. Just create topic for example `user-creation` which will be populated with the information and then create OpenFaaS function which handles the processing and updating of that information to the database.
+
+If we extend on the example above, again you would like to add some quick feature for the user.Attach another function to the same topic to extend the already existing functionality or just add another topic and write the function which handles that feature, it is as simple as that. 
+
+## Wrapping up
+
+Kafka Connector implements the [Connector SDK](https://github.com/openfaas-incubator/connector-sdk) with which you can create your own invoking mechanism.
+
+You can check the existing connectors like the [Cron Connector](https://github.com/zeerorg/cron-connector) or [vCenter Connector](https://github.com/openfaas-incubator/vcenter-connector) or all the existing function [Triggers](https://docs.openfaas.com/reference/triggers/).
+
+Now you already know the basics. It is over to you to deploy the connector with development tools like we did in the blog, or attach it to your existing Kafka deployment and start invoking functions. You can perhaps try attaching your function to multiple topics and extend the Hello World function with your own language of choice. Serverless with Kafka is all in your hands with the Kafka Connector.
```

Comment:
You cant attach a function to multiple topics yet

Reply:
I was hoping to merge this before the blog post is out?

Reply:
Removed the reference